kolibri/core/content/utils/transfer.py

```diff
@@ -163,9 +163,18 @@ def start(self):
         try:
             self.total_size = int(self.response.headers["content-length"])
         except Exception:
-            # HACK: set the total_size very large so downloads are not considered "corrupted"
-            # in importcontent._start_file_transfer
-            self.total_size = 1e100
+            # When a compressed file is saved on Google Cloud Storage,
+            # content-length is not available in the header,
+            # but we can use X-Goog-Stored-Content-Length.
+            gcs_content_length = self.response.headers.get(
+                "X-Goog-Stored-Content-Length"
+            )
+            if gcs_content_length:
+                self.total_size = int(gcs_content_length)
+            else:
+                # HACK: set the total_size very large so downloads are not considered "corrupted"
```

Comment:
@jonboiser Do you recall what would lead to this `corrupted` file result? I saw that in your peer-to-peer import PR, you added this hack to help resolve the issue of the corrupted files during peer-to-peer import. But I'm not able to reproduce it by running peer-to-peer locally.
I'm trying to do this because setting size to 1e100 now has some issue with the progress bar. Thank you!

Reply:
The `self.total_size = 1e100` hack was to make sure that field was set to very large number just in case the peer server was not setting its 'content-length' headers due to some kind of misconfiguration. It needed to be a large number since Kolibri would assume that file size being less than the 'content-length' header implied the file was corrupt.

Reply:
I forget how we first found this bug. I think if you try peer to peer with different localhost ports, it's fine. So maybe we tried doing peer to peer with either a kolibribeta server or kolibri studio itself, not sure.

Reply:
thank you for the clarification!